image_view/src/nodes/video_recorder.cpp

```diff
@@ -35,76 +37,112 @@ std::string encoding;
 std::string codec;
 int fps;
 std::string filename;
+std::string filename_orig;
 double min_depth_range;
 double max_depth_range;
 bool use_dynamic_range;
 int colormap;
+bool save_all_video;
 
+std::atomic<bool> recording(true);
 
 void callback(const sensor_msgs::ImageConstPtr& image_msg)
 {
-    if (!outputVideo.isOpened()) {
+    if (recording)
+    {
+        if (!outputVideo.isOpened()) {
 
-        cv::Size size(image_msg->width, image_msg->height);
+            cv::Size size(image_msg->width, image_msg->height);
 
-        outputVideo.open(filename, 
+            outputVideo.open(filename,
 #if CV_MAJOR_VERSION == 3
-                cv::VideoWriter::fourcc(codec.c_str()[0],
+                    cv::VideoWriter::fourcc(codec.c_str()[0],
 #else
-                CV_FOURCC(codec.c_str()[0],
+                    CV_FOURCC(codec.c_str()[0],
 #endif
-                          codec.c_str()[1],
-                          codec.c_str()[2],
-                          codec.c_str()[3]), 
-                fps,
-                size,
-                true);
-
-        if (!outputVideo.isOpened())
-        {
-            ROS_ERROR("Could not create the output video! Check filename and/or support for codec.");
-            exit(-1);
+                              codec.c_str()[1],
+                              codec.c_str()[2],
+                              codec.c_str()[3]),
+                    fps,
+                    size,
+                    true);
+
+            if (!outputVideo.isOpened())
+            {
+                ROS_ERROR("Could not create the output video! Check filename and/or support for codec.");
+                exit(-1);
+            }
+
+            ROS_INFO_STREAM("Starting to record " << codec << " video at " << size << "@" << fps << "fps. Press Ctrl+C to stop recording." );
+
         }
 
-        ROS_INFO_STREAM("Starting to record " << codec << " video at " << size << "@" << fps << "fps. Press Ctrl+C to stop recording." );
+        if ((image_msg->header.stamp - g_last_wrote_time) < ros::Duration(1.0 / fps))
+        {
+          // Skip to get video with correct fps
+          return;
+        }
 
+        try
+        {
+          cv_bridge::CvtColorForDisplayOptions options;
+          options.do_dynamic_scaling = use_dynamic_range;
+          options.min_image_value = min_depth_range;
+          options.max_image_value = max_depth_range;
+          options.colormap = colormap;
+          const cv::Mat image = cv_bridge::cvtColorForDisplay(cv_bridge::toCvShare(image_msg), encoding, options)->image;
+          if (!image.empty()) {
+            outputVideo << image;
+            ROS_INFO_STREAM("Recording frame " << g_count << "\x1b[1F");
```

Comment:
Can you please make this `ROS_DEBUG`? There's no need to spam the terminal every frame.